### ISSUE TITLE
Question -  How to deploy GCP  Second Gen cloud functions?

### ISSUE THREAD

**Issue by eduardoblucas** [serverless/serverless-google-cloudfunctions#289]
How to config the serverless.yaml to deploy second gen functions?

**Comment by pgrzesik** [comment]
Hello @eduardoblucas - I'm afraid that second gen functions are currently not supported.

**Comment by eduardoblucas** [comment]
@pgrzesik  Thank you, any idea if/when this will be implemented?

**Comment by faradaytrs** [comment]
I am sure many people would like to use it

**Comment by kevboutin** [comment]
> I am sure many people would like to use it

Absolutely! The benefits that second gen cloud functions brings puts it on par with Azure Cloud Functions and Lambdas. I would definitely hope to see this capability real soon.

**Comment by a88zach** [comment]
I can work on this as I also need this feature.  However, this feature is still in beta and is not supported by the v1 api endpoints.  Once the v2 api is out of beta,  I can create a PR

**Comment by sue445** [comment]
2nd gen is [already GA](https://cloud.google.com/blog/products/serverless/cloud-functions-2nd-generation-now-generally-available), and I could to find the v2 API in [googleapis](https://github.com/googleapis/google-api-nodejs-client/blob/main/src/apis/cloudfunctions/v2.ts)

**Comment by pgrzesik** [comment]
Hello folks, I’ve recently got access to this project and in the coming weeks I will try to investigate this issue (among a few other open initiatives). If you have any insights about this particular feature, please share it, it will be much appreciated 🙇

**Comment by pgrzesik** [comment]
Hey there, I've managed to dive a bit deeper into the differences between v1 and v2 and while the function definition is not that different, the major change is around supported events and how they're defined. It might require significant effort to handle that, so I'm not sure if I can provide any ETA for v2 support at the moment. I'll continue looking into it though 👍

**Comment by seanspeaks** [comment]
@pgrzesik Might be interested to help here- we're using serverless for our OS Frigg Framework (friggframework.org) and we've got a request for (and an obvious benefit to) using Gen2. I had peeked at it as well a few weeks ago and it looks ultimately like EITHER using v2 apis JUST for GCP, or re-writing everything to use v2 apis... which means a lot of type changes etc.

Any progress/thoughts on the approach?
